src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -154,18 +161,126 @@ void DirectoryService::ComputeSharding(
     {
         m_shards.emplace_back();
     }
+    map<array<unsigned char, BLOCK_HASH_SIZE>, PubKey> sortedPoWs;
+    vector<unsigned char> lastBlockHash(BLOCK_HASH_SIZE);
 
-    unsigned int i = 0;
+    if (m_mediator.m_currentEpochNum > 1)
+    {
+        lastBlockHash = HashUtils::SerializableToHash(
+            m_mediator.m_txBlockChain.GetLastBlock());
+    }
     for (const auto& kv : sortedPoWSolns)
     {
         const PubKey& key = kv.second;
-        map<PubKey, Peer>& shard = m_shards.at(i % numOfComms);
-        shard.emplace(key, m_allPoWConns.at(key));
-        m_publicKeyToShardIdMap.emplace(key, i % numOfComms);
+        const array<unsigned char, BLOCK_HASH_SIZE>& powHash = kv.first;
+
+        // sort all PoW submissions according to H(last_block_hash, pow_hash)
+        vector<unsigned char> hashVec;
+        hashVec.resize(BLOCK_HASH_SIZE + BLOCK_HASH_SIZE);
```

Comment:
For the `powHash` portion I think should use `POW_SIZE` even though technically it's the same as `BLOCK_HASH_SIZE`

Reply:
okay, noted.